lib/ex_plasma/crypto.ex

```diff
@@ -3,32 +3,46 @@ defmodule ExPlasma.Crypto do
   Signs and validates signatures. Constructed signatures can be used directly
   in Ethereum with `ecrecover` call.
   """
-  alias ExPlasma.Encoding
   alias ExPlasma.Signature
 
   @type sig_t() :: <<_::520>>
   @type pub_key_t() :: <<_::512>>
   @type address_t() :: <<_::160>>
   @type hash_t() :: <<_::256>>
 
+  @type recover_address_error() ::
+          :corrupted_witness
+          | :invalid_message
+
+  @doc """
+  Produces a KECCAK digest for the message.
+
+  see https://hexdocs.pm/exth_crypto/ExthCrypto.Hash.html#kec/0
+
+  ## Example
+
+    iex> ExPlasma.Crypto.keccak_hash("omg!")
+    <<241, 85, 204, 147, 187, 239, 139, 133, 69, 248, 239, 233, 219, 51, 189, 54,
+      171, 76, 106, 229, 69, 102, 203, 7, 21, 134, 230, 92, 23, 209, 187, 12>>
+  """
+  @spec keccak_hash(binary()) :: hash_t()
+  def keccak_hash(message), do: ExthCrypto.Hash.hash(message, ExthCrypto.Hash.kec())
+
   @doc """
   Recovers the address of the signer from a binary-encoded signature.
   """
-  @spec recover_address(hash_t(), sig_t()) :: {:ok, address_t()} | {:error, :signature_corrupt | binary}
+  @spec recover_address(hash_t(), sig_t()) :: {:ok, address_t()} | {:error, recover_address_error()}
   def recover_address(<<digest::binary-size(32)>>, <<packed_signature::binary-size(65)>>) do
     case Signature.recover_public(digest, packed_signature) do
       {:ok, pub} ->
         generate_address(pub)
 
-      {:error, "Recovery id invalid 0-3"} ->
-        {:error, :signature_corrupt}
-
-      other ->
-        other
+      {:error, _} ->
+        {:error, :corrupted_witness}
     end
   end
 
-  def recover_address(<<_digest::binary-size(32)>>, _signature), do: {:error, :invalid_signature}
```

Comment:
do these renamings affect other projects  that use ex_plasma?

Reply:
So, as we don't have it published as an official package, projects that use ex_plasma use it at a specific commit hash for now. With all the refactoring that has been done in this PR + some previous PR, projects that integrates it will surely need to refactor/rename things a bit when bumping to a newer commit. On the other hand, I also tried to do minimal changes to the external functions to make this process easier.